src/plugins/discover/public/application/components/sidebar/lib/visualize_url_utils.ts

```diff
@@ -104,7 +104,7 @@ export function getMapsAppUrl(
 
   return {
     app: 'maps',
-    path: `#/map?${mapAppParams.toString()}`,
+    path: `/map#?${mapAppParams.toString()}`,
   };
 }
```

Comment:
Code LGTM, Link works, however, when navigating back with the Browser's back button, it needs now 2 clicks, there seems to be a new redirect causing an additional history entry, think this should be checked

Reply:
I am seeing the same behavior.

The final URL is `http://localhost:5603/uhi/app/maps/map#?_g=(filters:!(),query:(language:kuery,query:''),refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))&_a=(filters:!(),query:(language:kuery,query:''))&initialLayers=(id:'207d55fd-a0ab-4414-9e99-301d6a2aa3aa',label:kibana_sample_data_logs,sourceDescriptor:(geoField:geo.coordinates,id:'8232074e-3b86-4366-9b27-ab18811eb6b8',indexPatternId:'90943e30-9a47-11e8-b64d-95841ca0b247',scalingType:CLUSTERS,tooltipProperties:!(),type:ES_SEARCH),type:BLENDED_VECTOR,visible:!t)`

Hitting the back button changes the URL to `http://localhost:5603/uhi/app/maps/map#?_g=(filters:!(),query:(language:kuery,query:''),refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))&_a=(filters:!(),query:(language:kuery,query:''))&initialLayers=(id:'207d55fd-a0ab-4414-9e99-301d6a2aa3aa',label:kibana_sample_data_logs,sourceDescriptor:(geoField:geo.coordinates,id:'8232074e-3b86-4366-9b27-ab18811eb6b8',indexPatternId:'90943e30-9a47-11e8-b64d-95841ca0b247',scalingType:CLUSTERS,tooltipProperties:!(),type:ES_SEARCH),type:BLENDED_VECTOR,visible:!t)`

Then hitting the back button a second time changes the url to `http://localhost:5603/uhi/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))&_a=(columns:!(_source),filters:!(),index:'90943e30-9a47-11e8-b64d-95841ca0b247',interval:auto,query:(language:kuery,query:''),sort:!())`

Reply:
Thanks @kertal. I've added a fix for this. Only one click backwards required now!